cleandart/react-dart
# UIP-2411: Basic DDC Support

# Ultimate problem

Consumers need `react-dart` to be compilable with the ddc.

# Solution

* Address ddc runtime errors that were not resolved as part of the strong mode changes.
* Add a new test page that renders both dom and custom components with no children, a list of children, and variadic children (currently commented out due to https://github.com/dart-lang/sdk/issues/29904).

# How to +10/QA
* Verify tests pass locally in Chrome (dart2js) and content_shell
    * `pub run test -p chrome`
    * `pub run test -p content-shell`
* Look at each of the test pages. None should have errors other than pre-existing key errors, expected key errors in `call_and_nosuchmethod_test.html` and an input related error in `react_test.html`.

---

please review: @greglittlefield-wf @jacehensley-wf @aaronlademann-wf
